### PR TITLE
Fix FixedCapacityDummyAllocator in CUDA memory pool test

### PR DESCRIPTION
**Pull request by niboshi**
Fixes the second problem of #5991

### CODE REVIEW

**Comment by hvy** [comment]
How about instead keep a dictionary of the sizes allocated by each pointer in the `FixedCapacityDummyAllocator` to simplify the logic?

**Comment by niboshi** [comment]
Fixed.

File: chainerx_cc/chainerx/cuda/memory_pool_test.cc [hunk 1]
```diff
@@ -116,20 +119,28 @@ class FixedCapacityDummyAllocator : public Allocator {
...
         return MallocStatus::kSuccess;
     }
     void Free(void* ptr) noexcept override {
-        intptr_t i = reinterpret_cast<intptr_t>(ptr);  // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
-        capacity_ += static_cast<size_t>(i);
-        ++free_called_;
+        uint8_t* mem = static_cast<uint8_t*>(ptr);
+        {
+            std::lock_guard<std::mutex> lock{sizes_mutex_};
+            capacity_ += sizes_.at(mem);
```
Comment (hvy) [inline chainerx_cc/chainerx/cuda/memory_pool_test.cc]: How about deleting the entry from `sizes_` after `Free`? Not sure if we need to add an existence check but I think that's logically more correct (avoiding it from keep growing).

**Comment by hvy** [comment]
Thanks for the fix! I added another small comment. Could you take a look?

File: chainerx_cc/chainerx/cuda/memory_pool_test.cc (hunk 1 above)
Comment (niboshi) [reply chainerx_cc/chainerx/cuda/memory_pool_test.cc]: Fixed.

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
LGTM!

**Pull request merged**